sn_registers/src/reg_crdt.rs

```diff
@@ -9,17 +9,21 @@
 use crate::{error::Result, Entry, EntryHash, Error, RegisterAddress, RegisterOp};
 
 use crdts::merkle_reg::Node as MerkleDagEntry;
-use crdts::{merkle_reg::MerkleReg, CmRDT, CvRDT};
+use crdts::{
+    merkle_reg::{Hash as CrdtHash, MerkleReg},
+    CmRDT, CvRDT,
+};
 use serde::{Deserialize, Serialize};
 use std::{
-    collections::BTreeSet,
+    collections::{BTreeSet, HashSet},
     fmt::{self, Debug, Display, Formatter},
     hash::Hash,
 };
+use xor_name::XorName;
 
 /// Register data type as a CRDT with Access Control
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Hash, PartialOrd)]
-pub(crate) struct RegisterCrdt {
+pub struct RegisterCrdt {
```

Comment:
Is there a specific motivation for making these items `pub`?

Reply:
I think that's needed so that the API can expose the MerkleReg. That's necessary for access to the history, and without it usefulness of Registers becomes very limited. Not useful really!

You can look at/use the `awe inspect-register` command and implementation for some of the things that are needed.

Reply:
Yes, we do want to expose the underlying register. But that would be the `MerkleReg` type. Our `RegisterCrdt` is not something we want to promise anything about for the future.